system/jlib/jfile.cpp

```diff
@@ -152,6 +152,25 @@ StringBuffer &swapPathDrive(StringBuffer &filename,unsigned fromdrvnum,unsigned
 }
 
 
+StringBuffer &getStandardPosixPath(StringBuffer &result, const char *path)
+{
+    result.set(path);
+    const char *s = result.trim().replace('\\', '/').str();
+    bool startWithPathSepChar = isPathSepChar(s[0]);
+    if (startWithPathSepChar)
+        s++;
+
+    if (*s && ((s[1]==':') || isShareChar(s[1])))
+    {
+        char c = tolower(s[0]);
+        if (!startWithPathSepChar)
+            result.insert(0, '/');
+        result.setCharAt(1, c);
+        result.setCharAt(2, '$');
+    }
+    return result;
```

Comment:
This is not time critical so may not matter that much.
But generally these type of routines scan through the input copying the input character or replacing on the fly, instead of copying, inserting and replacing.